connection.go

```diff
@@ -858,6 +858,14 @@ func (c *Connection) checkExchanges() {
 func (c *Connection) close(fields ...LogField) error {
 	c.log.WithFields(fields...).Debug("Connection closing.")
 
+	// Remaining frames in sendCh must be drained and released,
```

Comment:
It was nice that `writeFrames` was the only place that consumed from `sendCh`. I don't think this method is the right place since it can be called from many places. More details in the next comment.

Reply:
Hah, this was my idea - sorry @cinchurge. Seemed like "draining" as part of general shutdown was a reasonable thing to do. But maybe not :)